synapse/exc.py

```diff
@@ -217,6 +217,26 @@ class ModAlreadyLoaded(SynErr): pass
 class MustBeJsonSafe(SynErr): pass
 class NotMsgpackSafe(SynErr): pass
 
+class NoSuchForm(SynErr):
+
+    # new convention where exceptions with an init() class
+    # method may be called to enforce normalization of message
+    # contents based on variables that are relevant to the
+    # exception type...
+    @classmethod
+    def init(cls, name, mesg=None):
```

Comment:
We will need to ensure this design pattern survives pickling in the event that the exceptions are passed across multiprocessing pool.

Reply:
It only implements a ctor pattern that calls the regular one so it's no more/less of an issue than we already had :+1: